### PR TITLE
Spotfix/shortcode alias attributes

### PR DESCRIPTION
**Pull request by cliffordp**
demo of it working: https://share.getcloudapp.com/DOuQrkQ8

### CODE REVIEW

**Comment by sc0ttkclark** [comment]
Let's get a common ticket (for a future sprint) for this and find a way to work it into the array utils code instead.

**Comment by cliffordp** [comment]
@sc0ttkclark please give a fresh look, now moved to Array Utils

**Comment by cliffordp** [comment]
3min updated testing video: https://share.getcloudapp.com/bLueOgve

**Comment by sc0ttkclark** [comment]
Can we add at least a test for this in https://github.com/moderntribe/tribe-common/blob/spotfix/shortcode-alias-attributes/tests/wpunit/Tribe/Utils/ArrayTest.php?

File: src/Tribe/Shortcode/Shortcode_Abstract.php [hunk 1]
```diff
@@ -31,7 +31,16 @@ abstract class Shortcode_Abstract implements Shortcode_Interface {
...
 
 	/**
-	 * Array of callbacks for arguments validation
+	 * Account for this shortcode having aliased arguments.
+	 *
+	 * The array keys are aliases of the array values (i.e. the "real" shortcode attributes to allow parsing).
+	 *
+	 * @var string[]
+	 */
+	public $aliased_arguments = [];
```
Comment (bordoni) [inline src/Tribe/Shortcode/Shortcode_Abstract.php:40]: We should move this to a protected prop, and add a `set_*` and `get_*` into this.

File: src/Tribe/Shortcode/Shortcode_Abstract.php [hunk 2]
```diff
@@ -61,7 +70,8 @@ abstract class Shortcode_Abstract implements Shortcode_Interface {
 	 * {@inheritDoc}
 	 */
 	public function setup( $arguments, $content ) {
-		$this->arguments = $this->parse_arguments( $arguments );
+		$this->arguments = Arr::parse_associative_array_alias( (array) $arguments, (array) $this->aliased_arguments );
```
Comment (bordoni) [inline src/Tribe/Shortcode/Shortcode_Abstract.php:73]: Would love to see us move this call to the `parse_arguments`.

File: src/Tribe/Shortcode/Shortcode_Abstract.php (hunk 1 above)
Comment (bordoni) [reply src/Tribe/Shortcode/Shortcode_Abstract.php:40]: make sure the `set_*` and `get_*` are on the interface.

Comment (cliffordp) [reply src/Tribe/Shortcode/Shortcode_Abstract.php]: done

File: src/Tribe/Shortcode/Shortcode_Abstract.php (hunk 2 above)
Comment (cliffordp) [reply src/Tribe/Shortcode/Shortcode_Abstract.php]: great idea! done.

File: readme.txt [hunk 3]
```diff
@@ -2,9 +2,14 @@
...
 == Changelog ==
 
+= [4.12.2] 2020-05-19 =
+
+* Feature - Added array utility methods: `parse_associative_array_alias` to build an array with canonical keys while taking alias keys into account and `filter_to_flat_scalar_associative_array` to help do so. Useful for aliasing shortcode arguments, for example.
+
 = [4.12.1] 2020-05-11 =
 
 * Feature - Added a helper method `Tribe__Plugins::is_active( 'slug' )` to check if a given plugin is active.
+* Feature - Added array utility methods: `parse_associative_array_alias` to build an array with canonical keys while taking alias keys into account and `filter_to_flat_scalar_associative_array` to help do so. Useful for aliasing shortcode arguments, for example.
```
Comment (sc0ttkclark) [inline readme.txt:12]:
```suggestion
```

**Pull request merged**